src/main/java/net/atos/zac/aanvraag/ProductaanvraagService.java

```diff
@@ -270,15 +275,39 @@ public ProductaanvraagDimpact getProductaanvraag(final ORObject productaanvraagO
         }
     }
 
-    private void addInitiator(final String bsn, final URI zaak, final URI zaaktype) {
-        final RolType initiator = ztcClientService.readRoltype(RolType.OmschrijvingGeneriekEnum.INITIATOR, zaaktype);
-        final RolNatuurlijkPersoon rolNatuurlijkPersoon = new RolNatuurlijkPersoon(
-                zaak,
-                initiator,
-                ROL_TOELICHTING,
-                new NatuurlijkPersoon(bsn)
-        );
-        zrcClientService.createRol(rolNatuurlijkPersoon);
+    private Optional<Rol<?>> getRol(
+            final Betrokkene betrokkene,
+            final URI zaak,
+            final URI zaaktype,
+            RolType.OmschrijvingGeneriekEnum omschrijvingGeneriekEnum
+    ) {
+        final RolType initiator = ztcClientService.readRoltype(omschrijvingGeneriekEnum, zaaktype);
+        return getRol(betrokkene, zaak, initiator);
+    }
+
+    private Optional<Rol<?>> getRol(final Betrokkene betrokkene, final URI zaak, final RolType rolType) {
+        if (StringUtils.isNotBlank(betrokkene.getInpBsn()))
+            return Optional.of(new RolNatuurlijkPersoon(zaak, rolType, ROL_TOELICHTING, new NatuurlijkPersoon(betrokkene.getInpBsn())));
+        else if (StringUtils.isNotBlank(betrokkene.getVestigingsNummer()))
+            return Optional.of(new RolVestiging(zaak, rolType, ROL_TOELICHTING, new Vestiging(betrokkene.getVestigingsNummer())));
+        else if (StringUtils.isNotBlank(betrokkene.getInnNnpId()))
+            return Optional.of(
+                    new RolNietNatuurlijkPersoon(zaak, rolType, ROL_TOELICHTING, new NietNatuurlijkPersoon(betrokkene.getInnNnpId())));
+        else if (StringUtils.isNotBlank(betrokkene.getOrganisatorischeEenheidIdentificatie())) {
+            var eenheid = new OrganisatorischeEenheid();
+            eenheid.setIdentificatie(betrokkene.getOrganisatorischeEenheidIdentificatie());
+            return Optional.of(new RolOrganisatorischeEenheid(zaak, rolType, ROL_TOELICHTING, eenheid));
+        } else if (StringUtils.isNotBlank(betrokkene.getMedewerkerIdentificatie())) {
+            var medewerker = new Medewerker();
+            medewerker.setIdentificatie(betrokkene.getMedewerkerIdentificatie());
+            return Optional.of(new RolMedewerker(zaak, rolType, ROL_TOELICHTING, medewerker));
+        } else {
+            LOG.warning(String.format(
```

Comment:
Minor thing but I do wonder if this should just be a warning because to me it feels like a serious error somewhere.

Reply:
I made it severe.